### DISCUSSION TITLE
When re-running terraform apply a new bastion and operator are always created.

### DISCUSSION THREAD

**Discussion by codex70** [oracle-terraform-modules/terraform-oci-oke#470] [Q&A]
If I make no changes to anything, and re-run terraform apply, it tells me that bastion and sometimes the operator have been modified:
```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # module.bastion[0].oci_core_instance.bastion has changed
  ~ resource "oci_core_instance" "bastion" {
      + extended_metadata   = {}
        id                  = "ocid1.instance.oc1.eu-frankfurt-1.antheljrc6u6d5icsshkadmzomvep22pz4ixezx3yjqjj2w5lntesorzjglq"
        # (17 unchanged attributes hidden)
      ~ create_vnic_details {
          + nsg_ids                   = []
            # (9 unchanged attributes hidden)
        }
        # (7 unchanged blocks hidden)
    }

Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or respond to these changes.
```

Obviously this isn't ideal, but unfortunately it means that the extensions module isn't run, subsequently none of the applications are available on the operator. This causes terraform to get stuck in an endless loop waiting for applications to be deployed.

As a note, the bastion and operator are only available in a private address accessed via VPN.

Is there any work around for this, for example ignoring some changes, or forcing the extensions module to be re-run?

**Comment by boocup** [comment]
yes, i simulated a crash of the operator instance, ran terraform apply hoping a fully tooled operator would be created. I got the operator, but helm/kubectl where not installed. Shouldn't they be? @codex70 did you ever find a workaround?

**Reply by codex70** [reply]
Sorry for the late reply. There is now a fix for this to stop the operator and bastion getting created unnecessarily. Unfortunately, if the operator does need to be re-created, it looks like kubectl won't be reinstalled

**Reply by 12345ieee** [reply]
To fix that, all the null resources that install k8s tools need to gain a trigger of the type:

```tf
  triggers = {
    rerun_on_op_recreation = (some output of the operator module that changes each time the instance is rebuilt)
  }
  ```
  but I don't know if one has access to such details via the module, maybe with an extra datasource on top looking for instances with the right name.
